Makefile

```diff
@@ -124,6 +124,7 @@ test-containerized: run-etcd run-k8s-apiserver build-containerized dist/host-loc
 	-e PLUGIN=calico \
 	-e CNI_SPEC_VERSION=$(CNI_SPEC_VERSION) \
 	-v $(CURDIR):/go/src/github.com/projectcalico/cni-plugin:rw \
+	-v $(CURDIR)/dist:/opt/cni/bin \
```

Comment:
Hm.

It would be really nice if somehow we could instead just run the `calico/cni` container.

That might be more than we want to bite off right now, but I think it would simplify this a bit.  For example, we wouldn't have to [hack around in the test code](https://github.com/projectcalico/cni-plugin/pull/386/files#diff-2cac8d2716f994aa4216d877a9d2c84aR56) to make it look like the `calico/cni` container filesystem.

That's a more realistic test, and means we don't need to try to keep this test code and the live code in-sync, which is bug prone.

WDYT? Can you see a way to refactor so that we run the real container? It could be its own separate target - `make test-install-cni`.

Reply:
I was mostly trying to avoid a 2nd target for testing, but in reality I don't think it was necessary to do so. I don't see why this wouldn't work, I'll see if I can get that working.

Reply:
Yeah, a second target isn't the end of the world - `test-containerized` can just depend on it and it will be functionally the same.

Reply:
With running the install_cni container for each test then it might be possible to create a test that can't write one of the cni binaries.

Reply:
I tried installing `chattr` on Alpine, while it installs the filesystem is uses does not support the flags you can set with it so it simply errors out when you run it, so I'm still hunting for a way to make the files not writable by root.

Reply:
Turns out this isn't quite so simple, we don't have go installed in the `calico/cni` container, and cannot run Ginkgo, I can add it to the container if we feel it is worth it, WDYGT?

Reply:
Docker in docker?
I don't think it should be added.

Reply:
Definitely should not be added.

Maybe we need the test code itself to run on the local machine, but launch the `calico/cni` container via docker rather than running the script directly.

@ozdanborne recently did this for the FV tests in kube-controllers: https://github.com/projectcalico/kube-controllers/blob/master/Makefile#L122

Reply:
@heschlie do we still need this mount now?